### PR TITLE
fix palindrome products exercise

### PR DESCRIPTION
**Pull request by danielj-jordan**
#399

### CODE REVIEW

File: exercises/palindrome-products/example.js [hunk 1]
```diff
@@ -14,16 +14,16 @@ module.exports = function Palindromes(options) {
     for (var i = minFactor; i <= maxFactor; i++) {
       for (var j = minFactor; j <= maxFactor; j++) {
         var result = i * j;
-        if ( ! this.isPalindrome(result) ) { continue; }
+        if ( !this.isPalindrome(result) ) { continue; }
 
         var newFactor = [i, j].sort();
 
-        if (palindromes[result] === undefined) {
+        if ( !Array.isArray(palindromes[result])) {
```
Comment (matthewmorgan) [inline exercises/palindrome-products/example.js]: Hmm.  It seems like if there is a leading space, there should be a trailing one?  Or none at all?

**Comment by danielj-jordan** [comment]
@matthewmorgan I deleted the extra spaces inside the parentheses.   The linter did not give me an error for these spaces when they were present.  Did you run the linter and get an error?   I am curious if I have something misconfigured.

File: exercises/palindrome-products/example.js [hunk 2]
```diff
@@ -14,16 +14,16 @@ module.exports = function Palindromes(options) {
     for (var i = minFactor; i <= maxFactor; i++) {
       for (var j = minFactor; j <= maxFactor; j++) {
         var result = i * j;
-        if ( ! this.isPalindrome(result) ) { continue; }
+        if ( !this.isPalindrome(result) ) { continue; }
```
Comment (matthewmorgan) [inline exercises/palindrome-products/example.js]: Let's change this one too...

**Comment by matthewmorgan** [comment]
@danielj-jordan well, the linter default _should_ be to [disallow spaces inside parens](https://eslint.org/docs/rules/space-in-parens).  Just to be sure I checked the Airbnb docs, and they use the [default rule in this case](https://github.com/airbnb/javascript#whitespace--in-parens).

I tried running the linter locally and it didn't catch it.  I don't know why.

BTW there's one more place where there is extra space inside the parens, if you don't mind fixing that too, then we can merge your PR!

**Comment by matthewmorgan** [comment]
!! Thanks @danielj-jordan !!

**Pull request merged**